devchat.go

```diff
@@ -437,12 +438,52 @@ func (u *user) pickUsername(possibleName string) {
 		possibleName = cleanName(possibleName)
 	}
 	u.name = possibleName
-	u.changeColor(*colorArr[rand.Intn(len(colorArr))]) // also sets prompt
+	var colorIndex = rand.Intn(len(colorArr))
+	u.changeColor(colorNames[colorIndex], *colorArr[colorIndex]) // also sets prompt
 }
 
-func (u *user) changeColor(color color.Color) {
+func (u *user) changeColor(colorName string, color color.Color) {
 	u.name = color.Sprint(stripansi.Strip(u.name))
-	u.color = color
+	u.color = colorName
+	u.term.SetPrompt(u.name + ": ")
+	allUsersMutex.Lock()
+	allUsers[u.id] = u.name
+	allUsersMutex.Unlock()
+	saveBansAndUsers()
+}
+
+func (u *user) getColor() *color.Color {
```

Comment:
Might be a good idea to make a global variable that maps colors as strings to the actual color. (could replace this entire function)

Reply:
nope that's a bad idea :P